packages/wmr/src/lib/prerender.js

```diff
@@ -108,12 +87,47 @@ async function workerCode({ cwd, out, publicPath }) {
 		throw Error(`Unable to detect <script src="entry.js"> in your index.html.`);
 	}
 
+	/** @typedef {{ type: string, props: Record<string, string>, children?: string }} HeadElement */
+
+	/**
+	 * @type {{ lang: string, title: string, elements: Set<HeadElement>}}
+	 */
+	let head = { lang: '', title: '', elements: new Set() };
+	globalThis.wmr = { ssr: { head } };
+
 	// Prevent Rollup from transforming `import()` here.
 	const $import = new Function('s', 'return import(s)');
 	const m = await $import('file:///' + script);
 	const doPrerender = m.prerender;
 	// const App = m.default || m[Object.keys(m)[0]];
 
+	/**
+	 * @param {Set<HeadElement>} elements
+	 * @returns {Set<string>}
+	 */
+	function serializeHead(elements) {
+		return new Set(
+			Array.from(elements).map(e => {
+				if (typeof e !== 'string') {
+					const type = e.type;
+					let s = `<${type}`;
+					s += Object.keys(e.props)
+						.sort()
+						// Filter out empty values
+						.filter(p => e.props[p] != null)
+						.map(p => (p == 'children' ? '' : ` ${p}="${enc(e.props[p])}"`))
+						.join('');
+					s += '>';
+					let kids = e.props.children || e.children;
+					if (!Array.isArray(kids)) kids = [kids];
+					if (!/link|meta|base/.test(type)) s += `${kids.join('')}</${type}>`;
+					return s;
+				}
+				return e;
+			})
+		);
+	}
+
```

Comment:
Adding support for recursion (1/2)

```suggestion
	/**
	 * @param {HeadElement|HeadElement[]|Set<HeadElement>} element
	 * @returns {string} html
	 */
	function serializeElement(element) {
		if (element == null) return '';
		if (typeof element !== 'object') return String(element);
		if (element instanceof Set) element = Array.from(element);
		if (Array.isArray(element)) return element.map(serializeElement).join('');
		const type = element.type;
		let s = `<${type}`;
		const props = Object.keys(e.props).sort();
		let children = element.children;
		for (const prop of props) {
			const value = e.props[prop];
			// Filter out empty values
			if (value == null) continue;
			if (prop === 'children' || prop === 'textContent') children = value;
			s += ` ${p}="${enc(e.props[p])}"`;
		}
		s += '>';
		if (!/link|meta|base/.test(type)) {
			if (children) s += serializeElement(children);
			s += `</${type}>`;
		}
		return s;
	}
```